### PR TITLE
[dashboard] address lint warnings

### LINKED ISSUE #3841: [dashboard] Lint warnings
### Bug description
```console
Compiled with warnings.

src/App.tsx
  Line 57:8:   React Hook useEffect has a missing dependency: 'setUser'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
  Line 67:13:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images   jsx-a11y/alt-text

src/Login.tsx
  Line 26:9:    img elements must have an alt prop, either with meaningful text, or an empty string for decorative images   jsx-a11y/alt-text
  Line 71:8:    React Hook useEffect has a missing dependency: 'setUser'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
  Line 86:25:   img elements must have an alt prop, either with meaningful text, or an empty string for decorative images   jsx-a11y/alt-text
  Line 112:29:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images   jsx-a11y/alt-text
  Line 122:41:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images   jsx-a11y/alt-text

src/admin/UserSearch.tsx
  Line 41:8:  React Hook useEffect has a missing dependency: 'searchResult.rows'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

src/admin/WorkspacesSearch.tsx
  Line 51:8:  React Hook useEffect has a missing dependency: 'searchResult.rows'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
  Line 57:8:  React Hook useEffect has a missing dependency: 'search'. Either include it or remove the dependency array             react-hooks/exhaustive-deps

src/components/Menu.tsx
  Line 49:21:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images  jsx-a11y/alt-text

src/components/Modal.tsx
  Line 44:8:  React Hook useEffect has a missing dependency: 'handler'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

src/settings/EnvironmentVariables.tsx
  Line 29:45:   Unexpected whitespace after spread property operator  rest-spread-spacing
  Line 137:38:  Unnecessary escape character: \*                      no-useless-escape

src/settings/Integrations.tsx
  Line 48:8:    React Hook useEffect has a missing dependency: 'updateCurrentScopes'. Either include it or remove the dependency array                                                                                                                                                                                                                                                                                                                           react-hooks/exhaustive-deps
  Line 443:8:   React Hook useEffect has missing dependencies: 'props.mode', 'props.provider.host', 'props.provider.oauth.callBackUrl', 'props.provider.oauth.clientId', 'props.provider.oauth.clientSecret', and 'props.provider.type'. Either include them or remove the dependency array. If 'setType' needs the current value of 'props.provider.type', you can also switch to useReducer instead of useState and read 'props.provider.type' in the reducer  react-hooks/exhaustive-deps
  Line 447:8:   React Hook useEffect has a missing dependency: 'validate'. Either include it or remove the dependency array                                                                                                                                                                                                                                                                                                                                      react-hooks/exhaustive-deps
  Line 545:19:  Using target="_blank" without rel="noreferrer" is a security risk: see https://html.spec.whatwg.org/multipage/links.html#link-type-noopener                                                                                                                                                                                                                                                                                                      react/jsx-no-target-blank
  Line 546:13:  Using target="_blank" without rel="noreferrer" is a security risk: see https://html.spec.whatwg.org/multipage/links.html#link-type-noopener                                                                                                                                                                                                                                                                                                      react/jsx-no-target-blank
  Line 567:21:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images                                                                                                                                                                                                                                                                                                                                        jsx-a11y/alt-text
  Line 594:25:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images                                                                                                                                                                                                                                                                                                                                        jsx-a11y/alt-text
  Line 611:21:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images                                                                                                                                                                                                                                                                                                                                        jsx-a11y/alt-text
  Line 617:21:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images                                                                                                                                                                                                                                                                                                                                        jsx-a11y/alt-text

src/settings/Notifications.tsx
  Line 21:20:  Unexpected whitespace after spread property operator  rest-spread-spacing
  Line 22:24:  Unexpected whitespace after spread property operator  rest-spread-spacing
  Line 24:28:  Unexpected whitespace after spread property operator  rest-spread-spacing

src/settings/Plans.tsx
  Line 35:1:     Unexpected default export of anonymous function                                                                                                                                                                                                                                                                                                                   import/no-anonymous-default-export
  Line 73:8:     React Hook useEffect has missing dependencies: 'handleTeamClaim', 'pollAccountStatementTimeout', and 'server'. Either include them or remove the dependency array                                                                                                                                                                                                 react-hooks/exhaustive-deps
  Line 110:39:   Unexpected mix of '&&' and '||'                                                                                                                                                                                                                                                                                                                                   no-mixed-operators
  Line 110:81:   Unexpected mix of '&&' and '||'                                                                                                                                                                                                                                                                                                                                   no-mixed-operators
  Line 207:70:   Expected '===' and instead saw '=='                                                                                                                                                                                                                                                                                                                               eqeqeq
  Line 234:18:   Unexpected whitespace after spread property operator                                                                                                                                                                                                                                                                                                              rest-spread-spacing
  Line 253:70:   Expected '===' and instead saw '=='                                                                                                                                                                                                                                                                                                                               eqeqeq
  Line 274:18:   Unexpected whitespace after spread property operator                                                                                                                                                                                                                                                                                                              rest-spread-spacing
  Line 298:81:   The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid
  Line 298:127:  Script URL is a form of eval                                                                                                                                                                                                                                                                                                                                      no-script-url
  Line 321:81:   The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid
  Line 321:127:  Script URL is a form of eval                                                                                                                                                                                                                                                                                                                                      no-script-url
  Line 347:81:   The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid
  Line 347:127:  Script URL is a form of eval                                                                                                                                                                                                                                                                                                                                      no-script-url
  Line 396:21:   The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid
  Line 396:118:  Script URL is a form of eval                                                                                                                                                                                                                                                                                                                                      no-script-url
  Line 398:33:   The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid
  Line 398:85:   Script URL is a form of eval                                                                                                                                                                                                                                                                                                                                      no-script-url
  Line 399:29:   The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid
  Line 399:81:   Script URL is a form of eval                                                                                                                                                                                                                                                                                                                                      no-script-url
  Line 405:17:   img elements must have an alt prop, either with meaningful text, or an empty string for decorative images                                                                                                                                                                                                                                                         jsx-a11y/alt-text
  Line 406:129:  Using target="_blank" without rel="noreferrer" is a security risk: see https://html.spec.whatwg.org/multipage/links.html#link-type-noopener                                                                                                                                                                                                                       react/jsx-no-target-blank
  Line 413:25:   img elements must have an alt prop, either with meaningful text, or an empty string for decorative images                                                                                                                                                                                                                                                         jsx-a11y/alt-text
  Line 417:25:   img elements must have an alt prop, either with meaningful text, or an empty string for decorative images                                                                                                                                                                                                                                                         jsx-a11y/alt-text
  Line 430:25:   img elements must have an alt prop, either with meaningful text, or an empty string for decorative images                                                                                                                                                                                                                                                         jsx-a11y/alt-text
  Line 539:62:   Expected '===' and instead saw '=='                                                                                                                                                                                                                                                                                                                               eqeqeq

src/settings/Preferences.tsx
  Line 39:25:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images  jsx-a11y/alt-text
  Line 44:25:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images  jsx-a11y/alt-text

src/settings/Teams.tsx
  Line 108:8:    React Hook useEffect has a missing dependency: 'pollForAdditionalSlotsBought'. Either include it or remove the dependency array                     react-hooks/exhaustive-deps
  Line 154:8:    React Hook useEffect has missing dependencies: 'isChargebeeCustomer' and 'pendingPlanPurchase'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
  Line 162:8:    React Hook useEffect has a missing dependency: 'pendingSlotsPurchase'. Either include it or remove the dependency array                             react-hooks/exhaustive-deps
  Line 463:119:  Using target="_blank" without rel="noreferrer" is a security risk: see https://html.spec.whatwg.org/multipage/links.html#link-type-noopener         react/jsx-no-target-blank
  Line 576:25:   img elements must have an alt prop, either with meaningful text, or an empty string for decorative images                                           jsx-a11y/alt-text
  Line 605:8:    React Hook useEffect has a missing dependency: 'getPlan'. Either include it or remove the dependency array                                          react-hooks/exhaustive-deps
  Line 627:17:   img elements must have an alt prop, either with meaningful text, or an empty string for decorative images                                           jsx-a11y/alt-text
  Line 663:8:    React Hook useEffect has a missing dependency: 'plan.chargebeeId'. Either include it or remove the dependency array                                 react-hooks/exhaustive-deps
  Line 705:17:   img elements must have an alt prop, either with meaningful text, or an empty string for decorative images                                           jsx-a11y/alt-text
  Line 755:10:   'Slot' is already defined                                                                                                                           @typescript-eslint/no-redeclare
  Line 842:44:   Unexpected mix of '&&' and '||'                                                                                                                     no-mixed-operators
  Line 842:121:  Unexpected mix of '&&' and '||'                                                                                                                     no-mixed-operators
  Line 863:21:   img elements must have an alt prop, either with meaningful text, or an empty string for decorative images                                           jsx-a11y/alt-text

src/settings/settings-menu.ts
  Line 7:1:  Assign array to a variable before exporting as module default  import/no-anonymous-default-export

src/start/CreateWorkspace.tsx
  Line 270:6:   React Hook useEffect has a missing dependency: 'p.error.data'. Either include it or remove the dependency array                                                                                                                                                                               react-hooks/exhaustive-deps
  Line 301:21:  Assignments to the 'pollTimeout' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect  react-hooks/exhaustive-deps

src/start/StartPage.tsx
  Line 85:7:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images  jsx-a11y/alt-text

src/start/StartWorkspace.tsx
  Line 103:9:  Do not assign to the exception parameter                                                                                                  no-ex-assign
  Line 312:7:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images                                 jsx-a11y/alt-text
  Line 345:6:  React Hook useEffect has missing dependencies: 'logsEmitter' and 'props.workspaceId'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
  Line 373:6:  React Hook useEffect has missing dependencies: 'logsEmitter' and 'props.workspaceId'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

src/utils.ts
  Line 30:27:  Function declared in a loop contains unsafe references to variable(s) 'delayInSeconds'  no-loop-func

src/workspaces/StartWorkspaceModal.tsx
  Line 35:55:  React Hook useEffect has a missing dependency: 'computeSelection'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

src/workspaces/WorkspaceEntry.tsx
  Line 167:42:  Comparing to itself is potentially pointless  no-self-compare
  Line 173:41:  Comparing to itself is potentially pointless  no-self-compare
  Line 179:42:  Comparing to itself is potentially pointless  no-self-compare

src/workspaces/Workspaces.tsx
  Line 119:41:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images  jsx-a11y/alt-text
```

### Steps to reproduce

Requires changes from https://github.com/gitpod-io/gitpod/pull/3830
Run `craco build`

### Expected behavior

-

### Example repository

-

### PR DESCRIPTION
**Pull request by trumbitta**
## Description

The fixes are few and fairly easy so far, so I'm doing them all I guess.

Look at this beauty 💖
image

- Installed ESLint (and Prettier) VSCode extension
- Following the lead of the suggested fixes and working from there for non-basic ones

There are some TODOs I marked, which need clarifications/go-ahead by a couple Gitpodders:
I commented in the "code review" interface directly mentioning @geropl and @AlexTugarev. 

There are a couple more TODOs that mark linter warnings I silenced, with code that would need a more complex refactor.
Do you want me to open issues for each of them?

@jankeromnes I was thinking we could also add a linting step to the CI (I'm assuming there's none, at least for the dashboard, at this time) so that the warnings "per se" don't become errors but the CI fails when there are linting warnings.
I'm not familiar with the system in use, but I can look into it if you want.
To complete this, we could then add the files that still need work to [that `.eslintignore` you were suggesting](https://github.com/gitpod-io/gitpod/issues/3841#issuecomment-895253942).
Although I'd rather fix the linting warnings as soon as the problems arise (and fix everything in the dashboard right in this PR to start with a clean slate), because I fear files included in the `.eslintignore` would just become forgotten sooner than later 🙈 .

## Related Issue(s)

Fixes #3841 

## How to test
Use the dashboard 😬

Everything should still work, but warnings both in the terminal and in the browser devtools console should be gone for the files changed by this PR.

## Release Notes

```release-note
Fixed linting warnings in the dashboard app
```

### CODE REVIEW

File: components/dashboard/src/experiments.ts [hunk 1]
```diff
@@ -27,11 +27,18 @@ const Experiments = {
     /**
      * Experiment "example" will be activate on login for 10% of all clients.
      */
-    "example": 0.1,
+    example: 0.1,
 };
+
+// TODO: Both Experiment and Experiments are used for both code and typings here.
```
Comment (trumbitta) [inline components/dashboard/src/experiments.ts:33]: @geropl 🙏

File: components/dashboard/src/Menu.tsx [hunk 2]
```diff
@@ -47,256 +49,346 @@ export default function Menu() {
...
+        const all = [entry.link, ...(entry.alternatives || [])].map((l) => l.toLowerCase());
         const path = location.pathname.toLowerCase();
-        return all.some(n => n === path || n+'/' === path);
+        return all.some((n) => n === path || n + "/" === path);
     }
 
-    const userFullName = user?.fullName || user?.name || '...';
+    const userFullName = user?.fullName || user?.name || "...";
 
+    // TODO: This one smells like it could be some clever trickery,
```
Comment (trumbitta) [inline components/dashboard/src/Menu.tsx:71]: @AlexTugarev 🙏

**Comment by trumbitta** [comment]
No problem for the conflicts, I can redo everything but we need a strategy to avoid this from happening.

I can do micro-PRs, 1 file per PR, but I think we need some sort of agreement that the file would get a fairly quick review or we will be at this for the rest of the year :D

**Comment by trumbitta** [comment]
My latest proposal on how to avoid the merge conflicts and move forward https://discord.com/channels/816244985187008514/953232219261009950/953335514742788137

**Comment by trumbitta** [comment]
Following up from https://discord.com/channels/816244985187008514/953232219261009950/953335514742788137

I'll close this and I will start opening 1 PR per file 😊

**Comment by laushinka** [comment]
> I'll close this and I will start opening 1 PR per file 😊

Really appreciate these @trumbitta 🙏🏽

**Comment by trumbitta** [comment]
This is the first of the new smaller ones https://github.com/gitpod-io/gitpod/pull/8842

**Pull request closed without merging**